### PR TITLE
Raise `ImportError` when platform is not Linux

### PR DESCRIPTION
**Pull request by pentschev**
To prevent users inadvertently trying to use Dask-CUDA (and potentially other RAPIDS components) on a non-supported platform such as Windows, raise an `ImportError`.

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]

**Comment by pentschev** [comment]
Thanks @galipremsagar , let's just wait for @charlesbluca's confirmation that it doesn't break WSL2 before merging.

**Review by charlesbluca — APPROVED** [review]
Still able to import successfully on WSL2 🙂

**Comment by pentschev** [comment]
Thanks @charlesbluca for confirming!

**Pull request merged**
